### ISSUE TITLE
[BUG] VM appears twice on "Virtual Machines" webpage

### ISSUE THREAD

**Issue by ldevulder** [harvester/harvester#1111]
**Describe the bug**
When creating a new VM in `harvester-public` namespace I can see the VM twice on the `Virtual Machines` webpage:
![vm-twice-01](https://user-images.githubusercontent.com/25102165/130232562-554652db-c2d3-478e-87b0-724de3e7a248.png)
![vm-twice-02](https://user-images.githubusercontent.com/25102165/130232570-2cccae8b-7b58-4371-8959-546d2f547fba.png)

A refresh of the webpage doesn't change this. After some time only one line remains (but I don't know exactly how much time, after one hour for sure in my case).

**NOTE:** this issue is sporadic, I had it 3 times in 2 days  but not always...

**To Reproduce**
1. Create a new VM
2. Check if it appears twice

**Expected behavior**
The VM should appears only on one line.

**Support bundle**
I can't generate one, as there is a bug with this.

**Environment:**
 - Harvester ISO version: `master-3edc82a7-head`
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): HP ProLiant DL160 Gen9

**Additional context**

**Comment by WuJun2016** [comment]
@ldevulder  I am aware of a bug similar to this, but it will be normal after refreshing. If you encounter this problem again, please help me look at the returned data of the api.  thx

steps:
1. Enter https://ip:port/v1/harvester/kubevirt.io.virtualmachines
2. Compare id to see if it is duplicate

**Comment by yasker** [comment]
This should be obvious to observe during the testing. Close it as fixed now. Feel free to reopen if needed.

**Comment by WuJun2016** [comment]
This problem exists for all create operations:

Reproduction step: requires a slower network：
1. Create a secret or VM (without clicking the save button first),
2. open the browser console
3. Select Network
4. select "slow 3G" network
![image](https://user-images.githubusercontent.com/24985926/146419217-7045774b-b170-477b-bce5-6bbd670d6c88.png)
5.  click save

**Comment by gitlawr** [comment]
I've seen similar issue on Rancher v2.6.3-rc7 dashboard:
![image](https://user-images.githubusercontent.com/5697937/146496185-9f02cd71-0075-4644-a75c-7fa88839c8ee.png)

**Comment by richard-cox** [comment]
To help track down the cause of this, could whoever reproduces this do the following
- Check the browser's `Dev Tools` for any errors in the `Console`
- Check for the duplicate resources independently from the UI using `kubectl get` (the resource type is usually in the url, for instance the `Cluster Management` / `Clusters` page `c/local/manager/provisioning.cattle.io.cluster` --> `provisioning.cattle.io.cluster` in the `local` cluster)
- Confirm that the duplicate resources are still shown on refresh

For more details see https://github.com/rancher/dashboard/pull/4783#issuecomment-996216469

**Comment by richard-cox** [comment]
@MbolotSuse I heard on the grapevine that you're investigating a ghost cluster issue. I'm not sure if this is related (or possibly https://github.com/rancher/dashboard/issues/4949). There's some discussion in the associated PR https://github.com/rancher/dashboard/pull/4783, but basically it seems on save of a resource in the dashboard multiple versions of that resource are visible. On refresh they go away. I'd assume this is more of a frontend issue, but haven't reliably reproduced and investigated the socket messages

**Comment by MbolotSuse** [comment]
@richard-cox Yep I'm looking into it. Any data that I can get helps - thanks for the info. I'm not sure where the issue is quite yet. I'll let you know if there's anything related to the frontend that comes up.

**Comment by guangbochen** [comment]
@WuJun2016 probably can help to provide the detailed re-produce steps, thanks.
